### PR TITLE
Add building of RPMs and update default paths for DEB packages and command line parameters

### LINKED ISSUE #69: Build and publish RPM packages

### PR DESCRIPTION
**Pull request by soenkeliebau**
This PR makes a number of changes to the agent for both RPM and DEB packages, as well as default values for command line options.

I would have liked to split this into separate PRs, but as all these things are tied together there was no clean way of separating this, so I decided to bunch everything together.

Agent Code:
In line with [ADR011](https://github.com/stackabletech/documentation/blob/main/adr/ADR011-directory_structure.adoc) default value for the following parameters was changed (old value in [] for reference).
- data-directory: /var/lib/stackable/agent [/var/stackable/agent/data]
- bootstrap-file: /etc/stackable/stackable-agent/bootstrap-kubelet.conf [/etc/kubernetes/bootstrap-kubelet.conf]
- server-cert-file: /etc/stackable/stackable-agent/secret/agent.crt [None]
- server-key-file: /etc/stackable/stackable-agent/secret/agent.key [None]
- config-directory: /etc/stackable/serviceconfig [/opt/stackable/config]
- log-directory: /var/log/stackable/servicelogs [/opt/stackable/logs]

RPM:
Converted build process from operators to build agent rpm.

DEB:
Changed paths to be in line with the new defaults specified above.

Both packages currently do not deliver a config file, as they run completely of the default values. We should look into packaging a template config perhaps.

I am not entirely happy with the current directory structure for packaging, as we have some duplication in here, but I think it will serve for now and is not worth the effort needed to make it nicer (which would also make it more complex).

Also, changes to paths which files are installed into currently require a bit of digging and changes in multiple files - again, this is not nice, but I'm not sure it is worth the effort of making it nicer.

I've created issue #108 with some open questions that do not necessarily need to be included in this PR but should be addressed at some point.

### CODE REVIEW

**Comment by backstreetkiwi** [comment]
My review applies not to the way the packages are built here. Instead, as a user of the built artifacts I can say that the CentOS packages that are built by the action in this branch work in my tests with cluster generation in T2.

**Review by backstreetkiwi — DISMISSED** [review]

**Comment by siegfriedweber** [comment]
The paths described in the [documentation](https://github.com/stackabletech/agent/tree/764abb482dda7e699f731130d4faac916304ca2b) do not match with the ones configured in the rpm.

File: .rpm/systemd/stackable-agent.service [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+Description=Stackable Agent
+Before=
+After=network.target
+[Service]
+User=root
+ExecStart=/opt/stackable-agent/stackable-agent
+Restart=on-abort
+StandardOutput=journal
+StandardError=journal
+Environment="CONFIG_FILE=/etc/stackable-agent/agent.conf"
```
Comment (lfrancke) [inline .rpm/systemd/stackable-agent.service:11]: Is this correct?

File: packaging/rpm/SPECS/stackable-agent.spec [hunk 2]
```diff
@@ -0,0 +1,62 @@
+%define __spec_install_post %{nil}
```
Comment (lfrancke) [inline packaging/rpm/SPECS/stackable-agent.spec:1]: there are two stackable-agent.spec files. Which one is correct or are they both used?

**Review by lfrancke — COMMENTED** [review]
I think I have found issues but I'm not 100% sure. Please double check.

**Review by lfrancke — CHANGES_REQUESTED** [review]
See previous review

File: .rpm/stackable-agent.spec [hunk 3]
```diff
@@ -0,0 +1,57 @@
+%define __spec_install_post %{nil}
+%define __os_install_post %{_dbpath}/brp-compress
+%define debug_package %{nil}
+%define _bindir /opt/stackable-agent
+%define _confdir /etc/stackable-agent
```
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply .rpm/stackable-agent.spec:5]: This entire file is not needed anymore, I simply forgot to include deletion of the now unused .rpm folder in my commit.

File: .rpm/systemd/stackable-agent.service (hunk 1 above)
Comment (soenkeliebau) [reply .rpm/systemd/stackable-agent.service:11]: File is obsolete.

File: packaging/rpm/SPECS/stackable-agent.spec (hunk 2 above)
Comment (soenkeliebau) [reply packaging/rpm/SPECS/stackable-agent.spec:1]: Removed the incorrect one.

**Comment by soenkeliebau** [comment]
I have addressed the comments

> The paths described in the [documentation](https://github.com/stackabletech/agent/tree/764abb482dda7e699f731130d4faac916304ca2b) do not match with the ones configured in the rpm.

I'm not entirely sure if this comment still needs addressing after all the path changes I included in this PR?

**Review by lfrancke — APPROVED** [review]

**Pull request merged**